src/Cryptol/Eval/Reference.lhs

```diff
@@ -1332,8 +1347,10 @@ False]`, but to `error "foo"`.
 >              pure $ VRecord
 >                [ (f, go fty (lookupRecord f l') (lookupRecord f r'))
 >                | (f, fty) <- canonicalFields fs ]
->         TVNominal {} ->
->           evalPanic "arithBinary" ["Nominal type not in class `Ring`"]
+>         TVNominal _ _ ntv ->
+>           case ntv of
+>             TVStruct fs -> go (TVRec fs) l r
+>             _ -> evalPanic "arithBinary" ["Enum/Abstract type not in class `Ring`"]
```

Comment:
Similarly, this (and the other `evalPanic` cases) should mention `"ringBinary"` instead of `"arithBinary"` here.